### ISSUE TITLE
VF Language Server - textDocument/documentHighlight failed

### ISSUE THREAD

**Issue by humairazaman-devsinc** [forcedotcom/salesforcedx-vscode#5593]
### Summary

I have a VisualForce page. It has many tags, including apex tags, html tags and a script tag, and nothing goes wrong. The moment I add a <style> tag, every time I try to format it (and I have a format on save as well as an autosaver, so that happens a lot) I get the Visualforce Language Server error 
```
  Request textDocument/documentHighlight failed.
  Message: Request textDocument/documentHighlight failed with message: Debug Failure. False expression.
  Code: -32603 
```
The error message changes slightly according to unknown circumstances, I have also noticed `js_source_text is not defined`, but the error code remains the same.

If I comment out the script tag, everything works, and there is no error message.

EDIT: I just realized that `Debug Failure. False expression.` triggers if I click a line of code within the <script> tags, and the `js_source_text is not defined` shows up when I attempt to format. Neither shows up if the <style> tag is commented.

### Steps To Reproduce:

1.  Have a VisualForce page.
1.  Add a <style> tag.
2. Attempt to run the formatter.

### Expected result

Code is formatted.

### Actual result

Code is not formatted, instead, there is an error message in the Visualforce Language Server output.

### Additional information

<img width="1028" alt="Screen Shot 2024-05-10 at 8 43 01 AM" src="https://github.com/forcedotcom/salesforcedx-vscode/assets/139564731/c0abd858-242c-45ea-93e9-d5b56b219d11">

**Salesforce Extension Version in VS Code**: v60.12.0

**Salesforce CLI Version**: v60.12.0

**OS and version**: Mac v 12.7.4

**VS Code version**: Version: 1.89.0 (Universal)

**Most recent version of the extensions where this was working**: It never worked for me.

**Comment by daphne-sfdc** [comment]
Hi @humairazaman-devsinc, Can we see what other error messages are present?  What are the 5 errors in the Problems tab?  And what do you see when you run `SFDX: Toggle Developer Tools` in the command palette?

**Comment by humairazaman-devsinc** [comment]
The errors as pertains this page are:
<img width="811" alt="Screen Shot 2024-05-14 at 4 22 12 AM" src="https://github.com/forcedotcom/salesforcedx-vscode/assets/139564731/92c7a1e4-6bc7-4ad7-93cb-ec56080e1315">

I am not sure about any other messages, I believe there was at least one other message, but I was not paying attention at the time and I have no idea how to trigger it other than all this only happening after I added the style tag (to reformat the page, I have regularly been commenting out the style tag so it has been well proven to be the source of the problem).

`SFDX: Toggle Developer Tools` gave a No Matching Commands in the command palette, which offered `Developer: Toggle Developer Tools` as the closest alternative.

The SLDF extensions I am using are:
- Salesforce Extension Pack v60.12.0
- Salesforce Extension Pack (Expanded) v60.12.0

**Comment by humairazaman-devsinc** [comment]
https://github.com/forcedotcom/salesforcedx-vscode/assets/139564731/99657c82-ec7f-4d6c-a587-c5c5a7a7d93f

^ This is what happens on commenting/uncommenting the code, on formatting, on clicking the style line (I forgot to make it absolutely clear, but clicking something other than the style lines doesn't cause trouble) and on opening the dev tools.

I haven't used the dev tools before, so if there was something else you wanted to see, let me know and I'll check that too.

**Comment by daphne-sfdc** [comment]
@humairazaman-devsinc Thank you for your video of the issue.  Our team is currently working on an attempt to reproduce this issue.  We'll let you know when we have a reproduction and are able to proceed.  Thank you for your patience.

**Comment by daphne-sfdc** [comment]
Hi @humairazaman-devsinc, Our team has found a reproducer for this error and have identified this to be a bug.  We have created a new work item and will prioritize it in our next grooming session.

However, I'm still able to deploy the Visualforce page to my scratch org and see the results of the deployment after seeing the error.  Since the error only happens when you format, are you able to deploy if you turn off format on save?  If you save without formatting, are you able to deploy the Visualforce page and see the results?

**Comment by humairazaman-devsinc** [comment]
Yes, I can still deploy even with the error although I can't see the logs unless I preemptively go to another page beforehand, I can use other methods to format (or else comment the style tag, format, uncomment it).

There is no real blocker, only a major inconvenience.

Thanks for the help, and looking forward to having this fixed whenever it happens.

**Comment by fedegiust** [comment]
Any progress on this one? I'm having the same issue on Windows, this issue it seems that is preventing the Prettier formatter to be able to run.

![image](https://github.com/user-attachments/assets/5b4ff135-386b-444f-b26a-3f71af18ff94)

![image](https://github.com/user-attachments/assets/145d7a5d-ac10-40aa-a0ef-3db0194df707)

Prettier is configured for VF pages
![image](https://github.com/user-attachments/assets/36f92e20-dd5f-4b9f-8bfc-8fa3e9a658cb)

And the VF pages are not ignored
![image](https://github.com/user-attachments/assets/353a04db-b7bf-4c0e-96c3-82a12b837368)

**Comment by DanielTrontsios** [comment]
I am facing the same issue
